### PR TITLE
Swap out `ln` and `exp` for Solmate versions

### PR DESCRIPTION
**Pull request by TomAFrench**
# Description

This swaps out the exp and ln implementations for the solmate version. Draft as fuzzing of `pow` is tricky.

## Type of change

- [ ] Bug fix 
- [ ] New feature 
- [ ] Breaking change 
- [ ] Dependency changes
- [ ] Code refactor / cleanup
- [ ] Documentation or wording changes
- [ ] Other

## Checklist:

- [ ] The diff is legible and has no extraneous changes
- [ ] Complex code has been commented, including external interfaces
- [ ] Tests are included for all code paths
- [ ] The base branch is either `master`, or there's a description of how to merge

### CODE REVIEW

**Comment by TomAFrench** [comment]
An alternative to explore, we could use [PRBMath's](https://github.com/paulrberg/prb-math/blob/88d0815baef78c0699fec1ff10b34a35903e110f/contracts/PRBMathUD60x18.sol#L432-L438) implementation. It doesn't have the crazy optimisation theory maths but switching from using base e to base 2 should definitely save gas.

File: pkg/solidity-utils/test/foundry/LogExpMath.t.sol [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+        assertApproxEqRel(actualExp, expectedExp, 1e2);
+    }
+
+    function testLnEquivalence(int256 a) external {
+        vm.assume(a > 0);
+
+        int256 expectedLn = ReferenceLogExpMath.ln(a);
+        int256 actualLn = LogExpMath.ln(a);
+
+        assertApproxEqAbs(actualLn, expectedLn, 2);
```
Comment (nventuro) [inline pkg/solidity-utils/test/foundry/LogExpMath.t.sol:26]: Wow, I'm surprised this passes with such little difference.

File: pkg/solidity-utils/test/foundry/LogExpMath.t.sol [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+
+contract LogExpMathsTest is Test {
+    function testExpEquivalence(int256 a) external {
+        vm.assume(a > ReferenceLogExpMath.MIN_NATURAL_EXPONENT);
+        vm.assume(a < ReferenceLogExpMath.MAX_NATURAL_EXPONENT);
+
+        int256 expectedExp = ReferenceLogExpMath.exp(a);
+        int256 actualExp = LogExpMath.exp(a);
+
+        assertApproxEqRel(actualExp, expectedExp, 1e2);
```
Comment (nventuro) [inline pkg/solidity-utils/test/foundry/LogExpMath.t.sol:17]: It's good to test that they are quite close, but we also definitely want some ffi testing to test that the maximum error remains what it used to be.

**Review by nventuro — COMMENTED** [review]
I'm in favor of this renaming, but I'd do this in two steps. On the first PR, we create `NewLogExpMath` and rename the existing one to `ReferenceLogExpMath` (changing the imports as well). We have equivalence tests etc. Then, on a second PR (once we have confidence in the new impl), we rename `NewLogExpMath` to `LogExpMath`, and change the imports from `Reference` to `LogExpMath`. 

This may be slightly annoying as we'll need to touch the imports twice, but it clearly defines the point in which we change impls, and provides a clean diff (since we'll just have renames, unlike in this PR where the reference is a seemingly new file, and the new impl is also hard to review since the diff is messy (because it replaces old code).

**Comment by nventuro** [comment]
Have you performed any gas measurements to compare them?

**Comment by TomAFrench** [comment]
> Have you performed any gas measurements to compare them? Also bytecode size differences.

Not as of yet. My major concern is the removal of the 36 digit variant of `ln` so I was wanting to test `pow` against the TS implementation.
